authors/apps/authentication/views.py

```diff
@@ -10,7 +10,7 @@
 )
 
 
-class RegistrationAPIView(APIView):
+class RegistrationAPIView(GenericAPIView):
```

Comment:
Hello @Amoswachira ,
Did this task require changing the superclass for the view from APIView to GenericAPIView?

Reply:
@Ogutu-Brian the parameters in the swagger documentation were not appearing when using the APIView so i change it to GenericAPIView and The parameters  box appeared  where a user is required to input the JSON data .

Reply:
Does this imply that when I use APIView I will not be able to document my endpoint? We used APIViews before and we documented with swagger. Would you check what could be amiss?

Did you get a reason as to why the APIViews could not be documented but Generic views?

Reply:
@Ogutu-Brian This is a much efficient way as compared to what we had been doing in the past projects. when you use APIView you will have to create a separate file like swagger.py where you will have to add all the parameters manually which is tedious but when one uses GenericViews and Implements it this way. The parameters will be picked up automatically

Reply:
Hello @Amoswachira ,
Imagine situations, where APIViews are more efficient to use or another developer onboarding onto the codebase, is confident with APIViews, they might feel limited or devs might be limited on the classes they can use from a framework.

If introducing the swagger.py file would solve the problem now and in future, then I suggest that we go that way. This would provide a lasting solution

Reply:
@Ogutu-Brian it will not limit you. After what I demonstrated to you this morning i hope everything is fine now

Reply:
Yes it is